### PR TITLE
Re-load the migrated state and assert the payloads

### PR DESCRIPTION
**Pull request by SupunS**
Work towards https://github.com/onflow/cadence/issues/3143

Depends on https://github.com/onflow/flow-go/pull/5475

## Description

After writing the payloads back to the emulator state (store/db), reload the payloads from the state/store, and check whether the payloads are migrated properly.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: storage/migration/cadence1_test.go [hunk 1]
```diff
@@ -62,9 +78,285 @@ func TestCadence1Migration(t *testing.T) {
...
+				interpreter.NewReferenceStaticType(nil, interpreter.UnauthorizedAccess, rResourceType),
+			),
+		),
+
+		// TODO: Untyped Capability is not migrated?
+		//interpreter.NewUnmeteredCapabilityValue(
+		//	interpreter.NewUnmeteredUInt64Value(2),
+		//	interpreter.NewAddressValue(nil, address),
+		//	interpreter.NewReferenceStaticType(nil, interpreter.UnauthorizedAccess, rResourceType),
+		//),
```
Comment (SupunS) [inline storage/migration/cadence1_test.go:251]: I guess this should be fixed after https://github.com/onflow/cadence/pull/3125

**Review by turbolent — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

**Comment by SupunS** [comment]
There is an unrelated test `TestScriptWithCadenceRandom` failing after updating to latest flow-go (`stable-cadence` branch)

**Comment by turbolent** [comment]
@SupunS update again, flow-go `feature/stable-cadence` was broken, but it's fixed now

**Comment by janezpodhostnik** [comment]
@SupunS I hope you don't mind, I pushed a fix for the bootstrapping.

**Comment by SupunS** [comment]
@janezpodhostnik Thank you! I cherry-picked your changes to https://github.com/onflow/flow-emulator/pull/596

**Pull request merged**